pkg/ring/lifecycler.go

```diff
@@ -687,3 +752,38 @@ func (i *Lifecycler) unregister(ctx context.Context) error {
 		return ringDesc, true, nil
 	})
 }
+
+// TokenVersion1 is the version is a simple list of tokens.
+const TokenVersion1 = 1
+
+// Tokens is a simple list of tokens.
+type Tokens []uint32
+
+// MarshalTokens encodes given tokens into JSON.
+func MarshalTokens(tokens Tokens) ([]byte, error) {
```

Comment:
Any reason why `MarshalTokens` and `UnmarshalTokens` are not functions of the `Tokens` type (in that case should be just named `Marshal` and `Unmarshal`? Looking at the code base looks a common pattern.

Reply:
Looks like it can be a part of `Tokens`, will update.